### ISSUE TITLE
`//@ {unset-,}{rustc,exec}-env` parsing is a footgun

### ISSUE THREAD

**Issue by jieyouxu** [rust-lang/rust#132990]
Apparently

```
//@ rustc-env: RUSTC_BOOTSTRAP=1
```

is not the same as

```
//@ rustc-env:RUSTC_BOOTSTRAP=1
```

compiletest will parse the former as an env var called `⌴RUSTC_BOOTSTRAP` (incl. the whitespace).

**Comment by Shunpoco** [comment]
Hi, I'd like to take this ticket, but what is the best solution?
1. Accept such whitespace: I think we will modify Config::parse_env to remove such whitespaces
2. Fix all test codes which put a whitespace
3. Panic when parse_env detects such a whitespace

**Comment by jieyouxu** [comment]
I haven't quite determined what's the best solution to this yet.

**Comment by jieyouxu** [comment]
Tagging as E-hard and E-needs-design because it needs to not be a footgun on its own but also it needs to be homogenous with other directive syntaxes.

**Comment by Shunpoco** [comment]
Got it, thanks. Until the design is done, whitespace is clearly misconfiguration. Can I fix them?
(For example, I think we should remove whitespace from https://github.com/rust-lang/rust/blob/master/tests/ui/proc-macro/env.rs#L3)

**Comment by jieyouxu** [comment]
Yes, for the time being that sounds like a wise plan, I think the ones with whitespace in tests are probably wrong.
